plugin/kubernetes/kubernetes.go

```diff
@@ -102,10 +102,10 @@ func (k *Kubernetes) Services(ctx context.Context, state request.Request, exact
 
 		segs := dns.SplitDomainName(t)
 		if len(segs) != 1 {
-			return nil, nil
+			return nil, errInvalidRequest
 		}
 		if segs[0] != "dns-version" {
-			return nil, nil
+			return nil, errNoItems
```

Comment:
Strictly speaking, in the case that the TXT query is for a name that exists for another record type, the answer should be "NODATA".  This the proper DNS behavior.

So, before returning `errNoItems` (which results in NXDOMAIN), this code would need to check to see if a service with the name exists.  If one does exist, then the answer should be `nil` (which results in NODATA).

Reply:
Good point - I've got this working:
```
$ dig foo.svc.prod.cluster.local TXT @192.168.240.10 | grep status
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 49658

$ dig kubernetes.default.svc.prod.cluster.local TXT @192.168.240.10 | grep status
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 15694
```